packages/wallet/ui/src/service/Offers.js

```diff
@@ -1,36 +1,97 @@
+// @ts-check
+
 import {
   makeNotifierKit,
   makeAsyncIterableFromNotifier,
 } from '@agoric/notifier';
+import { E } from '@endo/eventual-send';
+
 import {
   loadOffers as load,
   removeOffer as remove,
   addOffer as add,
+  watchOffers,
 } from '../store/Offers.js';
 
+/** @typedef {import('@agoric/smart-wallet/src/types.js').Petname} Petname */
+
 /**
- * @param {string} publicAddress
+ * @param {string} chainId
+ * @param {string} address
  * @param {(data: string) => Promise<any>} signSpendAction
  * @param {Notifier<any>} chainOffersNotifier
+ * @param {Marshaller} marshaller
  */
 export const getOfferService = (
-  publicAddress,
+  chainId,
+  address,
   signSpendAction,
   chainOffersNotifier,
+  marshaller,
 ) => {
+  let pursePetnameToBrand;
   const offers = new Map();
-  let chainOffers = [];
   const { notifier, updater } = makeNotifierKit();
-  const broadcastUpdates = () =>
-    updater.updateState([...offers.values(), ...chainOffers]);
+  const broadcastUpdates = () => updater.updateState([...offers.values()]);
+
+  const augmentOffer = async offer => {
+    const {
+      id,
+      instanceHandleBoardId,
+      invitationMaker: { method },
+      proposalTemplate: { give, want },
+    } = offer;
+
+    const mapPurses = obj =>
```

Comment:
param should say what's expected, ideally with a type annotation. function name should say what comes out.

Reply:
Ok changed to `mapPursePetnamesToBrands`. Not exactly "what comes out" but more descriptive I think